--- a/arch/arm64/boot/dts/hisilicon/hi3660.dtsi	2026-09-07 23:03:16.746890805 +0200
+++ b/arch/arm64/boot/dts/hisilicon/hi3660.dtsi	2026-09-07 23:03:16.753245009 +0200
@@ -1271,6 +1271,35 @@
 				};
 			};
 		};
+
+		vdec: vdec@e8800000 {
+			compatible = "hisi,kirin960-vdec";
+			reg = <0x0 0xe8800000 0x0 0x20000>;
+			interrupts = <GIC_SPI 290 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 291 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 292 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 293 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 294 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 295 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 296 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 297 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&crg_ctrl HI3660_CLK_GATE_VDEC>;
+			clock-names = "clk_vdec";
+			dec_clk_rate = <480000000>, <200000000>;
+			status = "okay";
+		};
+
+		venc: venc@e8900000 {
+			compatible = "hisi,kirin960-venc";
+			reg = <0x0 0xe8900000 0x0 0x20000>;
+			interrupts = <GIC_SPI 298 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 299 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&crg_ctrl HI3660_CLK_GATE_VENC>,
+				 <&crg_ctrl HI3660_VENC_VOLT_HOLD>;
+			clock-names = "clk_venc", "venc_volt_hold";
+			enc_clk_rate = <645000000>, <534000000>;
+			status = "okay";
+		};
 	};
 };
 
