diff --git a/drivers-import/kirin960/dw_drm_dsi.c b/drivers-import/kirin960/dw_drm_dsi.c
index 16b2b92..f2b8d01 100644
--- a/drivers-import/kirin960/dw_drm_dsi.c
+++ b/drivers-import/kirin960/dw_drm_dsi.c
@@ -1641,7 +1641,7 @@ static struct platform_driver dsi_driver = {
 	.probe = dsi_probe,
 	.remove = dsi_remove,
 	.driver = {
-		.name = "dw-dsi",
+		.name = "kirin960-dw-dsi",
 		.of_match_table = dsi_of_match,
 	},
 };
