diff --git a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
index 6bd76c1fb007..dca4a9204be4 100644
--- a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
@@ -1449,8 +1449,8 @@ static const struct adv7511_chip_info adv7511_chip_info = {
 static const struct adv7511_chip_info adv7533_chip_info = {
 	.type = ADV7533,
 	.name = "ADV7533",
-	.max_mode_clock_khz = 80000,
-	.max_lane_freq_khz = 800000,
+	.max_mode_clock_khz = 148500,
+	.max_lane_freq_khz = 891000,
 	.supply_names = adv7533_supply_names,
 	.num_supplies = ARRAY_SIZE(adv7533_supply_names),
 	.reg_cec_offset = ADV7533_REG_CEC_OFFSET,
diff --git a/drivers/gpu/drm/bridge/adv7511/adv7533.c b/drivers/gpu/drm/bridge/adv7511/adv7533.c
index 188c1093a66e..da328c3c4f29 100644
--- a/drivers/gpu/drm/bridge/adv7511/adv7533.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7533.c
@@ -102,6 +102,13 @@ enum drm_mode_status adv7533_mode_valid(struct adv7511 *adv,
 {
 	struct mipi_dsi_device *dsi = adv->dsi;
 	u8 bpp = mipi_dsi_pixel_format_to_bpp(dsi->format);
+	unsigned int max_clk = (adv->num_dsi_lanes >= 4) ? 148500 : 80000;
+
+	if (mode->clock > max_clk)
+		return MODE_CLOCK_HIGH;
+
+	if (drm_mode_vrefresh(mode) < 48)
+		return MODE_BAD_VVALUE;
 
 	/* Check max clock for each lane */
 	if (mode->clock * bpp > adv->info->max_lane_freq_khz * adv->num_dsi_lanes)
