--- a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
+++ b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
@@ -595,7 +595,7 @@
 			compatible = "arm,pl011", "arm,primecell";
 			reg = <0x0 0xfdf01000 0x0 0x1000>;
 			interrupts = <GIC_SPI 57 IRQ_TYPE_LEVEL_HIGH>;
-			dmas = <&dma0 13 &dma0 13>;
 			dma-names = "rx", "tx";
 			clocks = <&crg_ctrl HI3660_UART4_FCLK>,
 				 <&crg_ctrl HI3660_PCLK>;
